### DISCUSSION TITLE
[FEATURE REQUEST]: Geospatial Support for streaming IOT device Data

### DISCUSSION THREAD

**Discussion by Yogesh-yrbyogi** [dotnet/spark#474] [Q&A] [answered]
**GeoSpatial Query Support for streaming IOT data **
For Finding the IOT Device Location based on the IOT Device Data, require the capabilities of the GeoSpatial Query during the streaming of the data.

**Describe the solution you'd like**
GeoSpatial query support while processing the stream data using the .Net Spark.

**Describe alternatives you've considered**
Using the Tile38 with streaming will provide support of the GeoSpatial.

**Comment by imback82** [comment]
@Yogesh-yrbyogi, can you be more specific about your ask? Do you have a library that works with Apache Spark for Geospatial and expose C# binding? Or something else?

**Comment by Yogesh-yrbyogi** [comment]
Dear @imback82 ,
Thanks for your quick update. I don't aware about the library & all. My requirement is to process millions of data IOT data & valid the each & every data against the Geo-fences (Predefined Location) created using the customized application.
I want to utilize the spark stream to process such big data, I require help for validating the IOT data with geospatial dataset.

**Comment by imback82** [comment, marked as answer]
I think you can try Spark's structured streaming with ForEach [API](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html#foreach) and C# examples can be found [here](https://github.com/dotnet/spark/blob/5f53809c1306d96ef7ffcf6004bd18ee380719d7/src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamWriterTests.cs#L203).

I am not familiar with the Geospatial, but maybe @MikeRys can help?
